src/components/NFTCollection/NFTCard.tsx

```diff
@@ -0,0 +1,158 @@
+// If image cannot be accessed using token_uri and use metadata
+const fromMetadata = (metadata: string | undefined) => {
+    let tempData = metadata && JSON.parse(metadata);
+    return {
+        name: tempData?.name,
+        image: tempData?.image,
+    };
+};
+
+const NFTCard: React.FC<NFTCardProp> = ({
+    metadata,
+    tokenAddress,
+    tokenUri,
+}) => {
+    const [nft, setNft] = useState<NFTType>({} as NFTType);
+    const [isError, setIsError] = useState(false);
+
+    const fetchData = async (url: string) => {
+        try {
+            const res = await fetch(url);
+            //If Link cannot be accessed(eg. opensea) try to display NFT from metadata
+            if (res.status != 200) {
+                const { name, image } = fromMetadata(metadata);
+                //if no image in metadata set error to true to not display this NFT
+                if (!name || !image) setIsError(true);
+                return { name, image };
+            } else {
+                const data = await res.json();
+                let ipfsLink = fixUrl(data.image);
+                return {
+                    name: data.name,
+                    image: ipfsLink ? ipfsLink : data.image,
+                };
+            }
+        } catch (error) {
+            //If there is any error in fetching image try from metadata
+            const { name, image } = fromMetadata(metadata);
+            if (!name || !image) setIsError(true);
+            return { name, image };
+        }
+    };
+
+    useEffect(() => {
+        if (tokenUri) {
+            const res = fetchData(tokenUri);
+            res.then((data) => setNft(data)).catch(() => setIsError(true));
+        }
+    }, [tokenUri]);
+
+    return nft.name ? (
+        <Row.Col
+            breakpointsConfig={{
+                xs: 24,
+                sm: 8,
+                md: 6,
+                lg: 6,
+            }}
+            data-testid="test-NFTCollection-id"
+            span={24}
+        >
+            <DivStyledCardContainer>
+                {/* On Image display error show lazyNft Illustration as default */}
+                {!isError ? (
+                    nft.image.slice(-3) === 'mp4' ? (
+                        <video
+                            height="200px"
+                            width="100%"
+                            src={nft.image}
+                            onError={() => setIsError(true)}
+                        ></video>
+                    ) : (
+                        <ImageStyled
+                            src={nft.image}
+                            onError={() => setIsError(true)}
+                        />
+                    )
+                ) : (
+                    <Illustration logo="lazyNft" width="100%" height="200px" />
+                )}
+                <DivStyledCardContent>
+                    <Typography variant="caption14">
+                        {getEllipsisTxt(tokenAddress, 4)}
+                    </Typography>
```

Comment:
instead of normal text you could make a link out of it which redirects you to the chain specific blockexplorer